XenModel/Actions/Pool/AddHostToPoolAction.cs

```diff
@@ -91,24 +92,31 @@ protected override void Run()
                 bool fixedCpus = FixCpus(Pool, _hostsToCpuMask, AcceptNTolChanges);
                 if (fixedCpus)
                     Session = NewSession();  // We've rebooted the server, so we need to grab the new session
-                RelatedTask = XenAPI.Pool.async_join(Session, Pool.Connection.Hostname, Pool.Connection.Username, Pool.Connection.Password);
+
+                var master = Pool.Connection.TryResolveWithTimeout(Pool.master);
+                var address = master != null ? master.address : Pool.Connection.Hostname;

```

Comment:
The following would I think be equivalent and arguably clearer:
`var address = master?.address ?? Pool.Connection.Hostname;`

Reply:
The two expressions are not exactly equivalent, namely `var address = master != null ? master.address : Pool.Connection.Hostname` will pass null into Pool.async_join if address is null, while `var address = master?.address ?? Pool.Connection.Hostname` will pass in the connection hostname. Not sure which behaviour is the best - the former will cause Pool.async_join to fail, but maybe the failure is desirable because the address shoud not be null. What do you think?